### PR TITLE
issue #1907 - strip the query from the originalRequstUri before parsing

### PR DESCRIPTION
**Pull request by lmsurpre**
Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 public class UriTest {
 
     @Test
-    public void testUriBadSecurity() throws URISyntaxException {
+    public void testUriTrimmedUrl() throws URISyntaxException {
```
Comment (prb112) [inline fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java:35]: hate to be a nit... let's update copyright year.

File: fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java [hunk 2]
```diff
@@ -72,7 +72,10 @@ public UriBuilder requestUri(String requestUriString) {
      * @throws URISyntaxException
      */
     public String toSearchSelfUri() throws URISyntaxException {
-        URI requestUri = new URI(requestUriString);
+        String hostAndPath = requestUriString.contains("?") ?
```
Comment (prb112) [inline fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java:75]: let's update copyright year.

File: fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java [hunk 3]
```diff
@@ -62,6 +62,37 @@ public void testUriBadSecurity() throws URISyntaxException {
             incoming);
     }
 
+    @Test
+    public void testUriBadSecurity() throws URISyntaxException {
+        String incoming =
+                "https://localhost:9443/fhir-server/api/v4/_search?_count=10&_security=http://ibm.com/fhir/security&_fudge=tag&_page=1";
+        String requestUriString = incoming;
```
Comment (prb112) [inline fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java:69]: maybe assign directly to requestUriString (however, it does model really well the existing code)

**Review by prb112 — APPROVED** [review]
LGTM, and makes sense based on what we discussed.

File: fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java:35]: its already 2021

File: fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java:75]: its already 2021

File: fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java (hunk 3 above)
Comment (lmsurpre) [reply fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java:69]: i left it as 2 different variables just for consistency with the other test.  to me, its more clear this way that one is testing a truncated URL (with no query part) and the other is testing the incoming url in whole

Comment (prb112) [reply fhir-search/src/test/java/com/ibm/fhir/search/uri/test/UriTest.java:69]: Yep, makes sense

**Comment by tbieste** [comment]
LGTM

**Pull request merged**